pkg/templates/node_1.10.go

```diff
@@ -41,6 +41,35 @@ systemd:
           contents: |
             [Service]
             Environment="DOCKER_OPTS=--log-opt max-size=5m --log-opt max-file=5 --ip-masq=false --iptables=false --bridge=none"
+    - name: flanneld.service
+      enable: true
+      dropins:
+        - name: 10-ccloud-opts.conf
+          contents: |
+            [Service]
+            ExecStartPre=/usr/bin/systemctl is-active ccloud-metadata-hostname.service
+            ExecStartPre=/usr/bin/bash -c "/usr/bin/systemctl set-environment NODE_NAME=$(hostname -s)"
+            Environment="FLANNEL_OPTS=-ip-masq=false \
+                                      -kube-subnet-mgr=true \
+                                      -kubeconfig-file=/var/lib/kubelet/kubeconfig \
+                                      -kube-api-url={{ .ApiserverURL }}"
+            Environment="RKT_RUN_ARGS=--uuid-file-save=/var/lib/coreos/flannel-wrapper.uuid \
+                                      --volume var-lib-kubelet,kind=host,source=/var/lib/kubelet,readOnly=true \
+                                      --mount volume=var-lib-kubelet,target=/var/lib/kubelet \
+                                      --volume etc-kubernetes-certs,kind=host,source=/etc/kubernetes/certs,readOnly=true \
+                                      --mount volume=etc-kubernetes-certs,target=/etc/kubernetes/certs \
+                                      --volume etc-kube-flannel,kind=host,source=/etc/kube-flannel,readOnly=true \
+                                      --mount volume=etc-kube-flannel,target=/etc/kube-flannel"
+    - name: flannel-docker-opts.service
```

Comment:
Is this an override,  so that flannel does nothing with docker?

Reply:
Yes, the `flannel-docker-opts.service` is locked in sync with `flanneld.service`. Both need to be able to start. It also wants to start before `docker` so that it can inject it's config. Without `docker` the `kubelet` won't start. If that doesn't start the node never registers itself and `flannel` doesn't find via the API and crashes. It's a nice dead-lock.

What I also learned is that systemd ignores a `restart=always` in a unit file when the service fails for any reason other than `Exec.*` fails with an error. Here it throws a dependency error and stops completely. Otherwise this convolution would maybe accidentally start eventually.

Anyway, we don't want the Docker options, so I'm overwriting it with a noop service. Flannel still crashes until the node registers itself. But now it actually restarts. Once the node is registered it also gets its `podCIDR`, so when Flannel comes up finally it's already set and reused.